HoloRepositoryDemoApplication/Assets/HoloRpository/Scripts/StorageConnectionServer.cs

```diff
@@ -0,0 +1,256 @@
+﻿using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.Networking;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using Microsoft.MixedReality.Toolkit.Utilities.Gltf.Serialization;
+using System.Collections.Generic;
+using System.Reflection;
+using SimpleJSON;
+
+namespace HoloRepository
+{
+    public class StorageConnectionServer : MonoBehaviour
+    {
+        #region Properties
+        private static string BaseUri = "http://localhost:3001/api/v1";
+        private static string WebRequestReturnData = null;
+        #endregion Properties
+
+        #region Public Method
+        public static void SetBaseUri(string Uri)
+        {
+            BaseUri = Uri;
+        }
+
+        public static IEnumerator GetMultiplePatient(List<Patient> patientList, string IDs)
+        {
+            //string MultiplePatientUri = BaseUri + "/patients?=" + "IDs";
+            string MultiplePatientUri = BaseUri + "/patients";
+            yield return GetRequest(MultiplePatientUri);
+
+            patientList.Clear();
+            if (WebRequestReturnData != null)
+            {
+                JSONNode InitialJsonData = JSON.Parse(WebRequestReturnData);
+                JSONArray JsonArray = InitialJsonData.AsArray;
+                foreach (JSONNode PatientJson in JsonArray)
+                {
+                    Patient patient = JsonToPatient(PatientJson);
+                    if (patient.pid != null)
+                    {
+                        patientList.Add(patient);
```

Comment:
I noticed that for all of your methods you have side-effects, so the result gets assigned/appended to an input argument. Do you have to do it that way in Unity? Is there no way to just work with regular method return values?

Reply:
There is no retrun value in the Coroutine, So this is the issue i mentioned before, the method will not wait for the web request, return value will always be null. I only found this solution so far